nemo/collections/nlp/modules/common/huggingface/huggingface_utils.py

```diff
@@ -33,12 +33,15 @@
 __all__ = ['MODELS', 'get_huggingface_lm_model', 'get_huggingface_lm_models_list']
 
 
-def get_huggingface_lm_model(pretrained_model_name: str, config_file: Optional[str] = None):
+def get_huggingface_lm_model(
+    pretrained_model_name: str, config_dict: Optional[dict], config_file: Optional[str] = None
```

Comment:
Shoudn't config_dict be =None by default?

Also, it seems to me that when both config_dict and config_file are specified then you should raise an exception saying that users should not do that.

Reply:
Good catch, I added the default.

I'm not sure about raising an exception. How about just a warning?  There is a precedence config from json > config from yaml > pretrained_model_name

Reply:
Okay, I added a warning if both config_dict and config_file are found. I also updated lm_model.py to notify the user when they are using a config_file.